[FIX][14.0][account_invoice_fixed_discount]: bad tax computation

flotho opened the pull request:
FIX #1107 for v14.0

Possibly same issue for triple discount

flotho committed:
[FIX][14.0]: bad tax computation

FIX #1107 for v14.0

more explicit variable

flotho commented:
ping @iggups , any chance to have a review here ?

flotho committed:
FIX pylint

FIX pylint

PYLINT.... at last

iggups commented:
@flotho looks good to me 👍

flotho commented:
hi @iggups you anly make a comment, any chance to make a positive review

iggups reviewed (approved):
Looks good!

flotho commented:
ping @OCA/accounting-maintainers ansy chance to be merged ?

adrienpeiffer commented:
Could you include a regression test showing the problem that this patch is solving and squash your commits ?

gaelTorrecillas committed:
fix discount fixed and pu =0.00

Update account_move.py

grindtildeath commented on account_invoice_fixed_discount/models/account_move.py:
```diff
@@ -71,7 +79,11 @@ def _get_price_total_and_subtotal_model(
     ):
         if self.discount_fixed != 0:
-            discount = ((self.discount_fixed) / price_unit) * 100 or 0.00
+            if price_unit >0.00:
```
You should use float tools functions such as `float_compare` or `float_is_zero` whenever comparing float values.

flotho replied:
thanks @grindtildeath 
@gaelTorrecillas could you please correct this.
thanks

flotho committed:
use float is zero

float comparison and defensive for non invoice documents

Improve conditions for computations

Defensive

flotho commented:
@grindtildeath , thanks for your comments, I've added the proper syntax
@adrienpeiffer I would love to add tests. What I need to test is when it comes from other part of the module Odoo, I mean I need test for data coming from sale.order and pos.order . How is it possible to make the module depnds on other only in test /demo mode ?

rafaelbn commented:
Quite interesting, don't miss the v16 migration https://github.com/OCA/account-invoicing/pull/1488 

#1488
